sunpy/map/mapbase.py

```diff
@@ -1995,6 +1995,21 @@ def _parse_submap_quantity_input(self, bottom_left, top_right, width, height):
     @_parse_submap_input.register(SkyCoord)
     @_parse_submap_input.register(BaseCoordinateFrame)
     def _parse_submap_coord_input(self, bottom_left, top_right, width, height):
+        # If bottom_left or top_right is a helioprojective coordinate with NaN values,
+        # cannot create meaningful submap
+        if (isinstance(bottom_left, Helioprojective)):
+            values = bottom_left.distance.to_value()
+            if np.isnan(values).any():
+              raise ValueError("The Helioprojective coordinates input to `submap()` contain NaN values."
```

Comment:
I think this message is the same as below? If so, I think defining it once above would be better.